src/openvic/pop/Pop.cpp

```diff
@@ -0,0 +1,107 @@
+#include "Pop.hpp"
+
+#include <cassert>
+
+#include "../utility/Logger.hpp"
+
+using namespace OpenVic;
+
+Pop::Pop(PopType const& new_type, Culture const& new_culture, Religion const& new_religion)
+	: type { new_type },
+	  culture { new_culture },
+	  religion { new_religion },
+	  size { 1 } {
+	assert(size > 0);
+}
+
+PopType const& Pop::get_type() const {
+	return type;
+}
+
+Culture const& Pop::get_culture() const {
+	return culture;
+}
+
+Religion const& Pop::get_religion() const {
+	return religion;
+}
+
+Pop::pop_size_t Pop::get_size() const {
+	return size;
+}
+
+PopType::PopType(std::string const& new_identifier, colour_t new_colour,
+	strata_t new_strata, sprite_t new_sprite,
+	Pop::pop_size_t new_max_size, Pop::pop_size_t new_merge_max_size,
+	bool new_state_capital_only, bool new_demote_migrant, bool new_is_artisan)
+	: HasIdentifier { new_identifier },
+	  HasColour { new_colour, true },
+	  strata { new_strata },
+	  sprite { new_sprite },
+	  max_size { new_max_size },
+	  merge_max_size { new_merge_max_size },
+	  state_capital_only { new_state_capital_only },
+	  demote_migrant { new_demote_migrant },
+	  is_artisan { new_is_artisan } {
```

Comment:
If you are making a bool for artisans, it might be good to also make `is_slave` since they have a variety of unique (non) interactions with things like voting and income.